### PR TITLE
Split handle_property into handle_sub and handle_set

### PR DESCRIPTION
**Pull request by laerling**
Matching https://github.com/vt6/vt6/pull/34

### CODE REVIEW

**Comment by laerling** [comment]
fixup!

**Comment by laerling** [comment]
Are all the lifetime annotations (`<'c>`) still needed?
Compiles well without them and tests still pass.

File: src/server/handler.rs [hunk 1]
```diff
@@ -159,7 +155,23 @@ pub trait Handler<C: Connection> {
     ///The return value shall be either `None` (if `name` is not a valid
     ///property or the required modules were not yet agreed to), or the return
     ///value from `MessageFormatter::publish_property()`.
-    fn handle_property<'c>(&self, name: &str, requested_value: Option<&[u8]>, conn: &mut C, send_buffer: &mut [u8]) -> Option<usize>;
+    fn handle_sub<'c>(&self, name: &str, conn: &mut C, send_buffer: &mut [u8]) -> Option<usize>;
```
Comment (majewsky) [inline src/server/handler.rs]: These function names don't reflect what they're about, and `sub` is ambiguous. I suggest:

- `handle_sub` -> `subscribe_to_property`
- `handle_set` -> `set_property`

File: src/server/core/handler.rs [hunk 2]
```diff
@@ -169,16 +180,29 @@ impl<C: Connection, H: server::Handler<C>> server::Handler<C> for Handler<H> {
...
+        } else {
+            self.next.handle_sub(name, conn, send_buffer)
+        }
+    }
+
+    fn handle_set<'c>(&self, name: &str, requested_value: &[u8], conn: &mut C, send_buffer: &mut [u8]) -> Option<usize> {
         use common::core::msg::prerecorded::publish_property;
-        //we do not support changing any properties yet, so just return the
-        //current value
+        //FIXME we do not support changing any properties yet, so just return
```
Comment (majewsky) [inline src/server/core/handler.rs]: That's not necessarily a FIXME. It just refers to us not having any connection-handling code that affords changing buffer sizes.

**Review by majewsky — COMMENTED** [review]
The lifetime annotations can go. Apart from that, you may merge when the review comments below have been addressed.

One thing that doesn't feel right is the code duplication that's going on in `handle_sub` and `handle_set` implementations, but we can deal with that later.

**Pull request merged**